File: hw/pastor/u3/hw28PrinciplesAplication/PYTHON/repository/repository_impl.py
from repository.i_repository import IRepository
from config.mongo_connection import MongoConnection
from model.generic_entity import GenericEntity

class RepositoryImpl(IRepository):
    def __init__(self, collection_name):
        self.db = MongoConnection().get_database()
        self.collection = self.db[collection_name]
        self.type_name = collection_name

    def create(self, entity):
        self.collection.insert_one(entity.to_dict())

    def read_all(self):
        cursor = self.collection.find()
        results = []
        for doc in cursor:
            entity = GenericEntity(self.type_name)
            entity.set_id(str(doc["_id"]))
            
            for key, value in doc.items():
                if key != "_id":
                    entity.set_data(key, value)
            
            results.append(entity)
        return results

File: hw/pastor/u3/hw28PrinciplesAplication/JAVASCRIPT/repository/repositoryImpl.js
const IGenericRepository = require('./iRepository');
const database = require('../config/database');
const GenericEntity = require('../model/genericEntity');

class RepositoryImpl extends IGenericRepository {
    constructor(collectionName) {
        super();
        this.collectionName = collectionName;
    }

    async create(entity) {
        const db = await database.connect();
        await db.collection(this.collectionName).insertOne(entity.toDocument());
    }

    async readAll() {
        const db = await database.connect();
        const docs = await db.collection(this.collectionName).find({}).toArray();
        
        return docs.map(doc => GenericEntity.fromDocument(doc, this.collectionName));
    }
}

module.exports = RepositoryImpl;